app/src/main/java/com/example/taskyy/di/AgendaModule.kt

```diff
@@ -26,7 +27,13 @@ object AgendaModule {
     @Provides
     @Singleton
     fun providesAgendaRepo(@ApplicationContext context: Context): AgendaRepository {
-        return AgendaRepositoryImpl(provideRetrofitInstance(context))
+        return AgendaRepositoryImpl(provideRetrofitInstance(context), provideUserDao(context))
+    }
+
+    @Provides
+    @Singleton
+    fun provideUserDao(@ApplicationContext context: Context): UserDao {
+        return AuthModule.provideRoomDatabase(context).userDao()
```

Comment:
Don't call these provide functions yourself, that completely breaks their lifecycle. Every time you call `provideRoomDatabase`, you'll create a new instance of it. Let Dagger handle this instead and don't call these yourself. To do that, just pass the dependencies as parameters here and Dagger will insert them:
```
fun provideUserDao(@ApplicationContext context: Context, db: TaskyyDatabase): UserDao {
    return db.userDao()
}
```